main/src/cgeo/geocaching/connector/gc/GCConstants.java

```diff
@@ -168,6 +168,13 @@
 
     static final String STRING_PREMIUMONLY_2 = "Sorry, the owner of this listing has made it viewable to Premium Members only.";
     static final String STRING_PREMIUMONLY_1 = "class=\"illustration lock-icon\"";
+    static final Pattern PATTERN_PREMIUMONLY_CACHETYPE = Pattern.compile("/app/ui-icons/sprites/cache-types\\.svg#icon\\-([^\"\\-]+)-?([^\"]+)?");
+    static final Pattern PATTERN_PREMIUMONLY_CACHENAME = Pattern.compile("<h1 class=\"heading\\-3\">([^<]+)");
```

Comment:
This will not work when ^ or < are part of the cache name.
I tested with "<" - this is possible.
What about `<h1 class="heading-3">(.+)<\/h1>` ?
(plus extra \ of cause)

Interesting question: is or was a cache name eg. `head </h1> or <h1> not?` possible?
At least today the html tags were filtered out on my tests creating such cache name.

Reply:
You are totally right! Fixed.